README.md

```diff
@@ -1,25 +1,33 @@
-### Fetch secrets from AWS Secrets Manager
+# Fetch secrets from AWS Secrets Manager
 
-#### Usage
+---
 
-This small binary fetches JSON formatted secrets from AWS Secrets Manager.  
-The secrets should be in a flat JSON K:V format. 
-The secrets path should usually follow the pattern `$service_name/$env/secrets`  
+## Usage
 
+This small binary fetches JSON formatted secrets from AWS Secrets Manager.  It adds the fetched secrets to the `env` for
+an binary that it executes after retrieving the secrets.
 
-It will use the the IAM role tags to autodiscover the secrets to be fetched, 
-any tags begginig with `secrets_` will be matched, the value should point to the secrets manager path.  
-For example the tag `secrets_default = myservice/dev/secrets` will load the secrets at that location. You can specify multiple tags, the merging order is arbitrary so don't expect precedence.
+The secrets should be stored in a AWS Secrets Manager secret with a value specified in a flat JSON K:V format.
+The Secrets Manager secret name should usually follow the path pattern `$service_name/$env/secrets`.  
 
 
-You then run it like this:  
-```sh
+`fetch-secrets` will use the IAM role tags to auto-discover the secrets to be fetched.
+Any tag keys prefixed with `secrets_` will be matched; the corresponding tag value be a Secrets Manager secret name.  
+eg; The tag `secrets_default = myservice/dev/secrets` will load the secrets stored in the secret named `myservice/dev/secrets`. 
+You can specify multiple tags, however, the merging order is arbitrary so don't expect precedence.
+
+---
+
+## Running
+
+```shell
 ./fetch-secrets mycommand subcommand --my-arg example
 ```
-This will load the variables and exec `mycommand` making them available as env variables in your application.
+This will load the variables and exec `mycommand` (with the subcommand and args), making the secret values available as env vars for the `mycommand` application.
 
-#### IAM configuration
-It will need the following policy to auto discover secrets:  
+###  IAM policy required for `fetch-secrets`
+
+In order to auto-discover secrets, `fetch-secrets` or the container/pod/instance it is running on requires the following IAM policy:  
 ```terraform
 policy = jsonencode({
```

Comment:
Given it's more advantageous to define the IAM policy in HCL (according to [the Terraform docs](https://developer.hashicorp.com/terraform/tutorials/aws/aws-iam-policy#refactor-your-policy)) perhaps you should whilst you're at it?

```terraform
resource "aws_iam_policy" "service" {
  name   = local.role_name
  policy = data.aws_iam_policy_document.service.json

  tags = {
    secrets_default = "${var.service_name}/${var.short_env}/secrets"
  }
}

data "aws_iam_policy_document" "service" {
  statement {
    actions = ["secretsmanager:GetSecretValue"]
    resources = [
      "arn:aws:secretsmanager:*:${data.aws_caller_identity.current.account_id}:secret:${var.service_name}/${var.short_env}/*"
    ]
    effect = "Allow"
  }

  statement {
    actions   = ["iam:ListRoleTags"]
    resources = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${local.role_name}"]
    effect    = "Allow"
  }

  statement {
    actions   = ["sts:GetCallerIdentity"]
    resources = ["*"]
    effect    = "Allow"
  }
}
```